content/post/articles/docker-basics-and-commands.md

```diff
@@ -0,0 +1,244 @@
+
+# Containers VS Virtual Machines
+
+Virtual machines allow multiple operating systems to run on the hardware of a single physical server, while containers allow multiple applications running the same operating system to be deployed on a single virtual machine or server.
+
+# What is Docker
+
+Docker is an open source platform used to create, run and deploy applications in containers. Docker allows you to separate applications from your infrastructure so you can deliver software quickly. With Docker, you can manage your infrastructure in the same way as your applications.
+
+## Image
+
+A Docker image is a file used to run code in a Docker container. Docker images act as a set of instructions for building a Docker container, like a template.
+
+Docker images are also the starting point for using Docker. An image is a template that contains a set of instructions for creating a container that can run on the Docker platform.
+
+A Docker image contains application code, libraries, tools, dependencies and other files needed to run an application. When a user runs an image, it can become one or more instances of a container.
+
+## Container
+
+Docker images and Docker containers are the two essential elements that we encounter every day when working with Docker.
+
+In simple terms, as explained above, a Docker image is a template that contains the application and all the dependencies required to run that application on Docker.
+
+On the other hand, a Docker container is a logical entity. In more precise terms, it is a running instance of the Docker image.
+
+A container is a standard software entity that bundles code and all its dependencies so that the application runs quickly and reliably from one environment to another. A Docker container image is a lightweight, self-contained, executable package of software that includes everything needed to run an application: code, runtime, system tools, system libraries, and settings.
+
+## Dockerfile
+
+The Dockerfile is a text document that contains all the commands a user could call on the command line to assemble a docker image.
+The Dockerfile is essentially the instructions for building the image. The advantage of a Dockerfile over simply storing the binary image (or a snapshot/template in other virtualisation systems) is that the automatic builds ensure that you have the latest version available.
+
+```bash
+# Example dockerfile to install Odoo
+# Note that this Odoo image is based on the debian:buster-slim image
+FROM odoo/odoo:14.0
+LABEL maintainer="You Name. <your@email.com>"
+
+COPY ./requirements_custom.txt /requirements_custom.txt
+
+RUN pip3 install --upgrade pip && pip3 install -r requirements_custom.txt
+
+WORKDIR /odoo/odoo
+
+USER odoo
+```
+
+## Docker-Compose
+
+Docker-Compose is a tool for defining and running multi-container Docker applications. With Docker-Compose, you use a YAML file to configure the services of your application before running it on docker.
```

Comment:
A bit of a nitpick here, but Docker-Compose shouldn't have an hyphen.

Reply:
thanks